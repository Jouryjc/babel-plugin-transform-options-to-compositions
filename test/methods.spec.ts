import plugin from '../lib/index'
import { transformSync } from '@babel/core'

describe('methods', () => {
  test('methods', () => {
    const { code } = transformSync(
      `
      export default {
        methods: {
          sayHello (aa) {
            console.log('say Hi')
            return aa
          }
        },
      }
    `,
      {
        presets: [
          [
            '@babel/preset-env',
            {
              modules: false,
            },
          ],
        ],
        plugins: [plugin],
      }
    )

    expect(code).toMatchSnapshot('methods')
  })
})
