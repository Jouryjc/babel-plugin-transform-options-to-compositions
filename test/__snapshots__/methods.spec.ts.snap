// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`methods methods: methods 1`] = `
"import 'vue';

const sayHello = aa => {
  console.log('say Hi');
  return aa;
};"
`;
